airbyte-integrations/bases/source-acceptance-tests/src/main/java/io/airbyte/integrations/acceptance_tests/source/FileBasedSourceAcceptanceTestDriver.java

```diff
@@ -0,0 +1,160 @@
+ */
+
+package io.airbyte.integrations.acceptance_tests.source;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.yaml.Yamls;
+import io.airbyte.integrations.acceptance_tests.source.core.CoreAcceptanceTest;
+import io.airbyte.integrations.acceptance_tests.source.core.CoreAcceptanceTestRunner;
+import io.airbyte.integrations.acceptance_tests.source.full_refresh.FullRefreshAcceptanceTestRunner;
+import io.airbyte.integrations.acceptance_tests.source.incremental.IncrementalAcceptanceTestRunner;
+import io.airbyte.integrations.acceptance_tests.source.models.CoreConfig;
+import io.airbyte.integrations.acceptance_tests.source.models.FullRefreshConfig;
+import io.airbyte.integrations.acceptance_tests.source.models.IncrementalConfig;
+import io.airbyte.integrations.acceptance_tests.source.models.SourceAcceptanceTestInputs;
+import io.airbyte.integrations.acceptance_tests.source.models.TestConfiguration;
+import io.airbyte.integrations.acceptance_tests.source.utils.TestRunner;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import net.sourceforge.argparse4j.ArgumentParsers;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+import net.sourceforge.argparse4j.inf.Namespace;
+
+public class FileBasedSourceAcceptanceTestDriver {
+
+  private static String readFileContents(String path) throws IOException {
+    return Files.readString(Path.of(path));
+  }
+
+  private static JsonNode deserializeJsonAtPath(String path, String entity) {
+    try {
+      return Jsons.deserialize(readFileContents(path));
+    } catch (Exception e) {
+      throw new RuntimeException("Could not deserialize " + entity + " at path " + path, e);
+    }
+  }
+
+  private static ConfiguredAirbyteCatalog deserializeConfiguredCatalogAtPath(String path) {
+    return deserializeObjectAtPath(path, ConfiguredAirbyteCatalog.class, "configured catalog");
+  }
+
+  private static <T> T deserializeObjectAtPath(String path, Class<T> klass, String entity) {
+    try {
+      return Jsons.deserialize(readFileContents(path), klass);
+    } catch (Exception e) {
+      throw new RuntimeException("Could not deserialize " + entity + " at path " + path, e);
+    }
+  }
+
+  List<Object> readTestInputs(SourceAcceptanceTestInputs testInputs) {
+    Preconditions.checkNotNull(testInputs.getTests(), "Input test suite configurations cannot be empty or null.");
+    Preconditions.checkArgument(testInputs.getTests().size() > 0, "Input test suite configurations cannot be empty or null.");
+    List<Object> testConfigs = Lists.newArrayList();
+    for (TestConfiguration testConfig : testInputs.getTests()) {
+      Object config = switch (testConfig.getType()) {
```

Comment:
could you just combine the switch statement in this method and the one in `runTestSuite` (technically a big conditional not a switch)? it would save you having to do the type casting. not sure that them being split is adding much.